Automattic/jetpack
# Use Storybook to publish Jetpack components

Pulls into #20691

This PR is a proposal for a way to browse and render Jetpack components, both as pure HTML and React.

The goal is to provide developers and designers with a playground for developing and using these components. 

It is based on a [very similar system](https://github.com/WordPress/gutenberg/tree/trunk/storybook) used by Gutenberg (though packaged slightly differently).

You can see the published (example) components at https://automattic.github.io/jetpack/storybook

#### Changes proposed in this Pull Request:
* Add a storybook tool to preview Jetpack UI components.

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:

To use:

```sh
$ cd projects/js-packages/storybook
$ pnpm install
$ pnpm run storybook:dev
```

This will build and open the Storybook UI in your default browser.

You should see an editor story (under "Playground") as well as a handful of components that already lived in the `@automattic/jetpack-components` directory.

Screenshot:

<img width="1211" alt="storybook-browser" src="https://user-images.githubusercontent.com/51896/129392702-ece8c57c-bb16-47ca-82f7-3fea7b9b0c8a.png">
